core/scripts/gateway/workflow/upload_workflow.go

```diff
@@ -0,0 +1,106 @@
+package main
+
+import (
+	"bytes"
+	"context"
+	"encoding/json"
+	"flag"
+	"fmt"
+	"io"
+	"net/http"
+	"os"
+
+	"github.com/ethereum/go-ethereum/crypto"
+	"github.com/joho/godotenv"
+
+	"github.com/smartcontractkit/chainlink/v2/core/services/gateway/api"
+)
+
+func main() {
+	gatewayURL := flag.String("gateway_url", "http://localhost:5002", "Gateway URL")
+	privateKey := flag.String("private_key", "65456ffb8af4a2b93959256a8e04f6f2fe0943579fb3c9c3350593aabb89023f", "Private key to sign the message with")
+	messageID := flag.String("message_id", "12345", "Request ID")
+	methodName := flag.String("method", "add_workflow", "Method name")
+	donID := flag.String("don_id", "workflow_don_1", "DON ID")
+	workflowSpec := flag.String("workflow_spec", "[my spec abcd]", "Workflow Spec")
```

Comment:
In real world scenario, this workflowSpec is the encoded combo of wasm binary and yaml config?

Reply:
Yes.